src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs

```diff
@@ -36,84 +30,23 @@ public static void Initialize()
-                }
-
-
-                // Initialize the timer, but don't set it to run.
-                // The timer will be set to run each time PollForTracingCommand is called.
-
-                // TODO: We should not need this timer once we are done settling upon a high-level design for
-                // what EventCounter is capable of doing. Once that decision is made, we should be able to 
-                // get rid of this.
-                if (_timer == null)
-                {
-                    _timer = new Timer(
-                        callback: new TimerCallback(PollForCounterUpdate),
-                        state: null,
-                        dueTime: Timeout.Infinite,
-                        period: Timeout.Infinite,
-                        flowExecutionContext: false);
-                }
-                // Trigger the first poll operation on when this EventSource is enabled
-                PollForCounterUpdate(null);
-            }
-            else if (command.Command == EventCommand.Disable)
-            {
-                if (_timer != null)
-                {
-                    _timer.Change(Timeout.Infinite, Timeout.Infinite);  // disable the timer from running until System.Runtime is re-enabled
-                }
-            }
-        }
-
-        public void UpdateAllCounters()
-        {
-            // GC counters
-            _counters[(int)Counter.GCHeapSize].WriteMetric(GC.GetTotalMemory(false));
-            _counters[(int)Counter.Gen0GCCount].WriteMetric(GC.CollectionCount(0));
-            _counters[(int)Counter.Gen1GCCount].WriteMetric(GC.CollectionCount(1));
-            _counters[(int)Counter.Gen2GCCount].WriteMetric(GC.CollectionCount(2));
-            _counters[(int)Counter.ExceptionCount].WriteMetric(Exception.GetExceptionCount());
-        }
-
-        private void PollForCounterUpdate(object state)
-        {
-            // TODO: Need to confirm with vancem about how to do error-handling here. 
-            // This disables to timer from getting rescheduled to run, which may or may not be 
-            // what we eventually want. 
-
-            // Make sure that any transient errors don't cause the listener thread to exit.
-            try
-            {
-                UpdateAllCounters();
-
-                // Schedule the timer to run again.
-                _timer.Change(EnabledPollingIntervalMilliseconds, Timeout.Infinite);
+                // NOTE: These counters will NOT be disposed on disable command because we may be introducing 
+                // a race condition by doing that. We still want to create these lazily so that we aren't adding
+                // overhead by at all times even when counters aren't enabled.
+
+                // On disable, PollingCounters will stop polling for values so it should be fine to leave them around.
+                _gcHeapSizeCounter = _gcHeapSizeCounter ?? new PollingCounter("GC Heap Size", this, () => GC.GetTotalMemory(false));
```

Comment:
In a future change we'll probably want to rename these counters so they don't have spaces and utilize the DisplayName property.